packages/perseus/src/widgets/categorizer/categorizer.tsx

```diff
@@ -15,31 +15,42 @@ import mediaQueries from "../../styles/media-queries";
 import sharedStyles from "../../styles/shared";
 import {getPromptJSON as _getPromptJSON} from "../../widget-ai-utils/categorizer/categorizer-ai-utils";
 
-import type {Widget, WidgetExports, WidgetProps} from "../../types";
+import type {
+    ChangeFn,
+    UniversalWidgetProps,
+    Widget,
+    WidgetExports,
+} from "../../types";
 import type {CategorizerPromptJSON} from "../../widget-ai-utils/categorizer/categorizer-ai-utils";
 import type {
     PerseusCategorizerWidgetOptions,
     PerseusCategorizerUserInput,
     CategorizerPublicWidgetOptions,
 } from "@khanacademy/perseus-core";
+import type {PropsFor} from "@khanacademy/wonder-blocks-core";
 
-type Props = WidgetProps<RenderProps> & {
-    values: ReadonlyArray<string>;
-};
+interface InternalProps extends UniversalWidgetProps {
+    values: number[];
+    items: PerseusCategorizerWidgetOptions["items"];
+    categories: PerseusCategorizerWidgetOptions["categories"];
+    randomizeItems: PerseusCategorizerWidgetOptions["randomizeItems"];
+}
 
-type DefaultProps = {
-    items: Props["items"];
-    categories: Props["categories"];
-    values: Props["values"];
-    linterContext: Props["linterContext"];
-};
+type ExternalProps = PropsFor<typeof Categorizer>;
+
+type RenderProps = Omit<ExternalProps, keyof UniversalWidgetProps>;
+
+type DefaultProps = Pick<
+    InternalProps,
+    "items" | "categories" | "values" | "linterContext"
+>;
```

Comment:
This is an idea that goes past this PR, but I wanted to note it. I suspect that for most widgets, the _only_ reason we use/need default props is to support usages in the editors. 

If we moved the default behaviour _into_ each widget's editor component, we could remove the defaulting of props from widgets entirely. Any defaulting at runtime would be handled by the parser and our types could become a bit simpler here.